### PR TITLE
Shawn

### PR DESCRIPTION
**Pull request by Feddas**
Can click items within a range of an item Collector.

### CODE REVIEW

File: GGJ2020/Assets/Scripts/Collectable.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    public Outline outline;
+
+    void Start()
+    {
+        outline = this.GetComponent<Outline>();
+    }
+
+    //void Update() { }
+
+    public void ClickItGood()
```
Comment (phelow) [inline GGJ2020/Assets/Scripts/Collectable.cs]: Confusing name

File: GGJ2020/Assets/Scripts/Collector.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        if (inCollectorRange == null || inCollectorRange.Count <= 0)
+            return;
+
+        // determine players mouse cursor position in world
+        var mouseInWorld = Input.mousePosition;
+        mouseInWorld.z = distanceOfCameraFromGround;
+        mouseInWorld = Camera.main.ScreenToWorldPoint(mouseInWorld);
+
+        // highlight closest object being hovered over by the mouse
+        var closestToMouse = inCollectorRange
```
Comment (phelow) [inline GGJ2020/Assets/Scripts/Collector.cs]: Is there an animation for when you click near something and pick it up.

File: GGJ2020/Assets/Scripts/Collector.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            }
+        }
+
+        if (inCollectorRange == null || inCollectorRange.Count <= 0)
+            return;
+
+        // determine players mouse cursor position in world
+        var mouseInWorld = Input.mousePosition;
+        mouseInWorld.z = distanceOfCameraFromGround;
+        mouseInWorld = Camera.main.ScreenToWorldPoint(mouseInWorld);
```
Comment (phelow) [inline GGJ2020/Assets/Scripts/Collector.cs:75]: This is really inefficient, I would rather see you use a circle cast for this.

File: GGJ2020/Assets/Scripts/Collector.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+        // highlight closest object being hovered over by the mouse
+        var closestToMouse = inCollectorRange
+            .OrderBy(c => Vector3.Distance(c.transform.position, mouseInWorld))
+            .First();
+
+        //Debug.Log(closestToMouse.name + " dist = " + Vector3.Distance(closestToMouse.transform.position, mouseInWorld));
+        if (Vector3.Distance(closestToMouse.transform.position, mouseInWorld) < mouseDistance)
```
Comment (phelow) [inline GGJ2020/Assets/Scripts/Collector.cs:83]: Circle cast by mouse distance instead

**Review by phelow — APPROVED** [review]
Next time give a more descritpive

File: GGJ2020/Assets/Scripts/Collector.cs (hunk 2 above)
Comment (Feddas) [reply GGJ2020/Assets/Scripts/Collector.cs]: There's an outline color change for all the clickablity states. Clicking it changes its outline to white for 0.5 seconds.

File: GGJ2020/Assets/Scripts/Collector.cs (hunk 3 above)
Comment (Feddas) [reply GGJ2020/Assets/Scripts/Collector.cs:75]: Fixed in new commit.

File: GGJ2020/Assets/Scripts/Collector.cs (hunk 4 above)
Comment (Feddas) [reply GGJ2020/Assets/Scripts/Collector.cs:83]: Only a few items are in the list this sorts for. Iterating that list avoids re-filtering if they are collectibles and in Line of Sight.

File: GGJ2020/Assets/Scripts/Collectable.cs (hunk 1 above)
Comment (Feddas) [reply GGJ2020/Assets/Scripts/Collectable.cs]: Got rid of the "Good" part.

**Pull request merged**